cluster/src/dunit/scala/io/snappydata/cluster/QueryRoutingDUnitSecureTest.scala

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package io.snappydata.cluster
+
+import java.sql.{BatchUpdateException, Connection, DriverManager, ResultSet, SQLException}
+
+import io.snappydata.test.dunit.AvailablePortHelper
+
+import org.apache.spark.Logging
+import org.apache.spark.sql.collection.Utils
+
+object QueryRoutingDUnitSecureTest{
+  val adminUser: String = "gemfire10"
+}
+
+class QueryRoutingDUnitSecureTest(val s: String)
+    extends ClusterManagerLDAPTestBase(s,
+      QueryRoutingDUnitSecureTest.adminUser) with Logging {
+
+  override def setUp(): Unit = {
+    super.setUp()
+  }
+
+  override def tearDown2(): Unit = {
+    super.tearDown2()
+  }
+
+  def netConnection(netPort: Int, user: String, pass: String): Connection = {
+    val driver = "io.snappydata.jdbc.ClientDriver"
+    Utils.classForName(driver).newInstance
+    val url: String = "jdbc:snappydata://localhost:" + netPort + "/"
+    DriverManager.getConnection(url, user, pass)
+  }
+
+  def insertRows1(numRows: Int, serverHostPort: Int, tableName: String,
```

Comment:
Can we have better names functions for example insertUsingBatches etc. instead of insertRows1, insertRows2 etc.?